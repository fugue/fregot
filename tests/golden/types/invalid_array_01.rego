package invalid_array_01

array_bools = ["Hello", "world"]
array_strings = ["Fire", "the", "missiles"]

test_array_concat {
  array_big = array.concat(array_bools, array_strings)
  array_big[_] > 0
}

test_array_slice {
  array_small = array.slice(array_strings, 1, 2)
  array_small[_] > 0
}

test_set_intersection {
  set_bool_string = {true, "Hello"}
  set_string_bool = {"Hello", false}
  set_string = intersection({set_bool_string, set_string_bool})
  set_string[_] > 0
}
