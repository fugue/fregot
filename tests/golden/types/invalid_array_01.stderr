fregot (typecheck error):
  "invalid_array_01.rego" (line 8, column 3):
  Subtype error:

    8|   array_big[_] > 0
         ^^^^^^^^^^^^^^^^

  Cannot use type string as number

fregot (typecheck error):
  "invalid_array_01.rego" (line 13, column 3):
  Subtype error:

    13|   array_small[_] > 0
          ^^^^^^^^^^^^^^^^^^

  Cannot use type string as number

fregot (typecheck error):
  "invalid_array_01.rego" (line 20, column 3):
  Subtype error:

    20|   set_string[_] > 0
          ^^^^^^^^^^^^^^^^^

  Cannot use type string|boolean as number
